fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -2030,520 +2023,181 @@ class ShardManager {
-	TraceEvent e(SevInfo, "ShardedRocksDBMetrics", debugID);
-	uint64_t stat;
-	for (auto& [name, ticker, cumulation] : tickerStats) {
-		stat = stats->getTickerCount(ticker);
-		e.detail(name, stat - cumulation);
-		cumulation = stat;
-	}
-	for (auto& [name, property] : intPropertyStats) {
-		stat = 0;
-		ASSERT(db->GetAggregatedIntProperty(property, &stat));
-		e.detail(name, stat);
+		intPropertyStats = {
+			{ "NumImmutableMemtables", rocksdb::DB::Properties::kNumImmutableMemTable },
+			{ "NumImmutableMemtablesFlushed", rocksdb::DB::Properties::kNumImmutableMemTableFlushed },
+			{ "IsMemtableFlushPending", rocksdb::DB::Properties::kMemTableFlushPending },
+			{ "NumRunningFlushes", rocksdb::DB::Properties::kNumRunningFlushes },
+			{ "IsCompactionPending", rocksdb::DB::Properties::kCompactionPending },
+			{ "NumRunningCompactions", rocksdb::DB::Properties::kNumRunningCompactions },
+			{ "CumulativeBackgroundErrors", rocksdb::DB::Properties::kBackgroundErrors },
+			{ "CurrentSizeActiveMemtable", rocksdb::DB::Properties::kCurSizeActiveMemTable },
+			{ "AllMemtablesBytes", rocksdb::DB::Properties::kCurSizeAllMemTables },
+			{ "ActiveMemtableBytes", rocksdb::DB::Properties::kSizeAllMemTables },
+			{ "CountEntriesActiveMemtable", rocksdb::DB::Properties::kNumEntriesActiveMemTable },
+			{ "CountEntriesImmutMemtables", rocksdb::DB::Properties::kNumEntriesImmMemTables },
+			{ "CountDeletesActiveMemtable", rocksdb::DB::Properties::kNumDeletesActiveMemTable },
+			{ "CountDeletesImmutMemtables", rocksdb::DB::Properties::kNumDeletesImmMemTables },
+			{ "EstimatedCountKeys", rocksdb::DB::Properties::kEstimateNumKeys },
+			{ "EstimateSstReaderBytes", rocksdb::DB::Properties::kEstimateTableReadersMem },
+			{ "CountActiveSnapshots", rocksdb::DB::Properties::kNumSnapshots },
+			{ "OldestSnapshotTime", rocksdb::DB::Properties::kOldestSnapshotTime },
+			{ "CountLiveVersions", rocksdb::DB::Properties::kNumLiveVersions },
+			{ "EstimateLiveDataSize", rocksdb::DB::Properties::kEstimateLiveDataSize },
+			{ "BaseLevel", rocksdb::DB::Properties::kBaseLevel },
+			{ "EstPendCompactBytes", rocksdb::DB::Properties::kEstimatePendingCompactionBytes },
+			{ "BlockCacheUsage", rocksdb::DB::Properties::kBlockCacheUsage },
+			{ "BlockCachePinnedUsage", rocksdb::DB::Properties::kBlockCachePinnedUsage },
+			{ "LiveSstFilesSize", rocksdb::DB::Properties::kLiveSstFilesSize },
+		};
 	}
 
-	int64_t manifestBytesTotal = 0;
-	auto manifests = getManifestBytes(manifestDirectory);
-	int idx = 0;
-	for (const auto& [fileName, fileBytes] : manifests) {
-		e.detail(format("Manifest-%d", idx), format("%s-%lld", fileName.c_str(), fileBytes));
-		manifestBytesTotal += fileBytes;
-		idx++;
+	std::vector<std::pair<std::string, int64_t>> getManifestBytes(std::string manifestDirectory) {
+		std::vector<std::pair<std::string, int64_t>> res;
+		std::vector<std::string> returnFiles = platform::listFiles(manifestDirectory, "");
```

Comment:
If listFiles throw platform_error, can the outside capture this error? or the logger silently exits?

Reply:
This is not a retryable error. It will propagate to worker causing storage server reboot.